migration/v0.30/apply/14-hnc-labels.sh

```diff
@@ -5,40 +5,68 @@ ROOT="$(readlink -f "$(dirname "${0}")/../../../")"
 # shellcheck source=scripts/migration/lib.sh
 source "${ROOT}/scripts/migration/lib.sh"
 
-update_hnc_namespaces_label() {
-  log_info "--- Waiting for hnc-controller-manager"
-  kubectl_do wc wait pods -n hnc-system -l app.kubernetes.io/component=hnc-controller-manager --for condition=Ready --timeout=90s
-
-  readarray userNamespaces < <(yq4 '.user.namespaces[]' $CK8S_CONFIG_PATH/wc-config.yaml)
-  readarray userNamespacesWithHigherPSALevel < <(yq4 '.user.constraints | keys | .[]' $CK8S_CONFIG_PATH/wc-config.yaml)
+update_hierarchyconfigurations() {
+  readarray -t userNamespaces < <(yq4 '.user.namespaces[]' "$CK8S_CONFIG_PATH/wc-config.yaml")
+  readarray -t userNamespacesWithHigherPSALevel < <(yq4 '.user.constraints // {} | keys | .[]' "$CK8S_CONFIG_PATH/wc-config.yaml")
 
   for i in "${userNamespacesWithHigherPSALevel[@]}"; do
-    userNamespaces=(${userNamespaces[@]//*$i*/})
+    userNamespaces=("${userNamespaces[@]//*${i}*/}")
   done
 
   for i in "${userNamespaces[@]}"; do
-    ## Trigger the new mutation on already existing hnc namespaces
-    kubectl_do wc get hierarchyconfigurations -n ${i} hierarchy -o yaml |
-      yq4 '.metadata.generation += 1' |
-      kubectl_do wc apply -f -
+    if [[ -z "$(kubectl_do wc get hierarchyconfigurations.hnc.x-k8s.io -n "${i}" hierarchy --ignore-not-found)" ]]; then
+      continue
+    fi
+
+    kubectl_do wc annotate hierarchyconfigurations.hnc.x-k8s.io -n "${i}" hierarchy helm.sh/hook- helm.sh/resource-policy- meta.helm.sh/release-name=user-rbac meta.helm.sh/release-namespace=kube-system
+    kubectl_do wc label hierarchyconfigurations.hnc.x-k8s.io -n "${i}" hierarchy app.kubernetes.io/managed-by=Helm
+
+    config="$(kubectl_do wc get hierarchyconfigurations.hnc.x-k8s.io -n "${i}" hierarchy -oyaml)"
+
+    kubectl_do wc patch hierarchyconfigurations.hnc.x-k8s.io -n "${i}" hierarchy --type merge --patch '{
+      "spec": {
+        "labels": [
+          {
+            "key": "pod-security.kubernetes.io/enforce",
+            "value": "restricted"
+          },
+          {
+            "key": "pod-security.kubernetes.io/audit",
+            "value": "restricted"
+          },
+          {
+            "key": "pod-security.kubernetes.io/warn",
+            "value": "restricted"
+          }
+        ]
+      }
+    }'
   done
 }
 
 run() {
   case "${1:-}" in
   execute)
-    log_info "--- applying user-rbac for workload cluster"
-    helmfile_upgrade wc app=user-rbac
-
     log_info "--- applying HNC for workload cluster"
     helmfile_upgrade wc group=hnc
 
```

Comment:
```suggestion
    log_info "--- applying HNC for workload cluster"
    helmfile_upgrade wc group=hnc

    sleep 10
```

To prevent the wait to trigger on old pods.
Or do a double wait and catch if the first errors out.

Reply:
Added the wait
Please recheck